lib/libyear_bundler/cli.rb

```diff
@@ -2,29 +2,22 @@
 require "bundler/cli/outdated"
 require "libyear_bundler/report"
 require "libyear_bundler/query"
+require "libyear_bundler/options"
 
 module LibyearBundler
   # The `libyear-bundler` command line program
   class CLI
-    OPTIONS = %w[
-      --all
-      --grand-total
-      --libyears
-      --releases
-      --versions
-    ].freeze
-
     E_BUNDLE_OUTDATED_FAILED = 1
     E_NO_GEMFILE = 2
 
     def initialize(argv)
-      @argv = argv
+      @options = ::LibyearBundler::Options.new(argv).parse!
+      @argv = argv # `parse_argv` removes non-flag options, leaving e.g. a Gemfile path
       @gemfile_path = load_gemfile_path
-      validate_arguments
     end
 
     def run
-      if @argv.include?("--grand-total")
+      if @options[:grand_total?]
```

Comment:
Is `@options.grand_total?` possible? That would be easier to read.

Reply:
We could use a struct for `@options`.